sys/events.js

```diff
@@ -85,6 +85,26 @@ EventService.prototype._emit = function(hyper, req) {
                 return event;
             })
             .filter(function(event) { return !!event; });
+
+            // Change-propagation will set up the x-rerender-reason header, indicating
+            // the event which caused the rerender. In case RESTBase is about to emit
+            // the same event, it will cause a rerender loop. So, log an error and skip
+            // an event.
+            var reRenderReason = req.headers && req.headers['x-rerender-reason']
+                || hyper._rootReq && hyper._rootReq['x-rerender-reason'];
+            if (reRenderReason) {
+                events = events.filter(function(event) {
+                    if (reRenderReason.replace(/^https?:/, '') === event.meta.uri.replace(/^https?:/, '')) {
```

Comment:
So, as a conclusion, we all agree we need to get the optional XFF-like field in meta in all events. 

As for the regex - basically we're making sure that a loose `s` in https doesn't get us in trouble. Events in the extension are emitted the https, while restbase exists them with http.